src/test/java/com/google/cloud/spanner/pgadapter/ITDdlTest.java

```diff
@@ -72,25 +72,36 @@ public void testDropTableWithIndex() throws SQLException {
     try (Connection connection = DriverManager.getConnection(getConnectionUrl())) {
       // First create a table with a secondary index.
       try (Statement statement = connection.createStatement()) {
-        statement.addBatch("create table test (id bigint primary key, value varchar)");
-        statement.addBatch("create index idx_text on test (value)");
-        assertArrayEquals(new int[] {0, 0}, statement.executeBatch());
+        statement.addBatch("create table test1 (id bigint primary key, value varchar)");
+        statement.addBatch("create table test2 (id bigint primary key, value varchar)");
+        statement.addBatch("create index idx_text1 on test1 (value)");
+        statement.addBatch("create index idx_text2 on test2 (value)");
+        assertArrayEquals(new int[] {0, 0, 0, 0}, statement.executeBatch());
       }
 
-      // Try to drop the table with an index. This will fail.
-      PSQLException exception =
-          assertThrows(
-              PSQLException.class, () -> connection.createStatement().execute("drop table test"));
-      assertEquals(SQLState.FeatureNotSupported.toString(), exception.getSQLState());
-      assertNotNull(exception.getServerErrorMessage());
-      assertEquals(
-          "Execute 'set spanner.support_drop_cascade=true' to enable dropping tables with indices",
-          exception.getServerErrorMessage().getHint());
-
-      // Now enable drop_cascade.
-      connection.createStatement().execute("set spanner.support_drop_cascade to on");
-      // Dropping the table should now work.
-      assertEquals(0, connection.createStatement().executeUpdate("drop table test"));
+      for (int index = 1; index <= 2; index++) {
+        connection.createStatement().execute("set spanner.support_drop_cascade to off");
+
+        String tableName = "test" + index;
+        boolean useIfExists = index == 2;
+        String dropStatement = "drop table " + (useIfExists ? "if exists " : "") + tableName;
```

Comment:
This will always fail because there is no "cascade" keyword.

Reply:
The `spanner.support_drop_cascade` enables the standard cascade behavior of PG. That means:
1. Dropping a table with an index works automatically (also without `cascade`, as this is the default in PG).
2. Dropping a table with other dependencies work if you include the `cascade` keyword.

So that means that the following does work (and if it didn't, then this test would have failed):

```
set spanner.support_drop_cascade to on;
create table foo (id bigint primary key, value varchar);
create index idx_foo on foo (value);
drop table foo;
```

Reply:
I see. An index is not like a foreign key constraint. Thanks for the explanation.